plugin/abolish.vim

```diff
@@ -29,21 +29,21 @@ endfunction
 
 function! s:send(self,func,...)
   if type(a:func) == type('') || type(a:func) == type(0)
-    let Func = get(a:self,a:func,'')
+    let l:Func = get(a:self,a:func,'')
   else
-    let Func = a:func
+    let l:Func = a:func
   endif
   let s = type(a:self) == type({}) ? a:self : {}
-  if type(Func) == type(function('tr'))
-    return call(Func,a:000,s)
```

Comment:
I bet you can fix the issue by changing this line and this line alone. Can you confirm/refute?

Reply:
I can confirm that scoping only the first four occurrences of `Func` with `l:` fixes the issue. Do you want me to push another commit with only those changes and leave the other `Func` alone?

Reply:
I'm sorry, I don't know much about git or github, so when you say "this line", I'm not sure what line I should change.
Just above your comment I can see these 2 lines in red:

    if type(Func) == type(function('tr'))
      return call(Func,a:000,s)

So, I assume you want me to only scope the first 4 occurrences of `Func` like this:

    if type(a:func) == type('') || type(a:func) == type(0)
      let l:Func = get(a:self,a:func,'')
    else
      let l:Func = a:func
    endif
    if type(l:Func) == type(function('tr'))
      return call(l:Func,a:000,s)

If that's what you meant, then yes, I've just made a test and it fixes the issue.

Do you want me to push another commit and only change these 4 occurrences, and leave the 10 other ones alone?

Reply:
Just the `return call(Func,a:000,s)` line is what I was thinking. If that works then yes, I'd favor the single line diff.

Reply:
Ok, I've just made a test, the only change being this line:

    return call(Func,a:000,s)

Replaced by this line:

    return call(l:Func,a:000,s)

But unfortunately, the issue still persists.

Reply:
Oh, it must be the 2 assignment lines.

Reply:
I agree, I think the error comes the assignments, that's why I scoped them with `l:` in the commit.
And since I did that for the assigments, I felt it was necessary to go on and do the same for all the other occurrences, otherwise I thought that `Func` would refer to the user-defined function, instead of the local funcref.

If I only scope the assignments, when I press `cru`, something weird happens (the word is deleted, or the cursor moves and a character is deleted somewhere else); it depends on what the current line contains...

However, as I said in the previous post, if I scope only the first four occurrences, then this issue is fixed.
So if you want me to scope only them, and leave the other alone, I'll do it and push another commit.

Although, I'm not sure a similar issue would not happen if `s:send()` had to process the following `elseif` statements where `Func` is unscoped.
FWIW, I always scope a funcref with `l:` to avoid having headaches trying to understand what Vim will understand. But I admit I have far less experience in VimL than you, so if it's fine by you, it's fine by me.

Reply:
I favor the minimum possible diff which appears to be the 2 assignment lines. Since you closed this I'll take care of it. Thanks for the recon!